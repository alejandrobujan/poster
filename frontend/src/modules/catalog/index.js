import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


