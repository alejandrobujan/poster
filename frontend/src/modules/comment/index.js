import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};