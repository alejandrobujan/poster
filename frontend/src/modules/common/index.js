export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";