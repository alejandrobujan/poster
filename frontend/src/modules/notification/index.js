import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Notifications} from "./components/Notifications";
export {default as Notification} from "./components/Notification";

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


