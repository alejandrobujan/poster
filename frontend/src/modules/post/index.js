import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as CreatePost} from './components/CreatePost';
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';
export {default as PostDetails} from './components/PostDetails';


// eslint-disable-next-line
export default  {actions, reducer, selectors};