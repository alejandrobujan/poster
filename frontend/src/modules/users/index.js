import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Login} from './components/Login';
export {default as SignUp} from './components/SignUp';
export {default as Logout} from './components/Logout';
export {default as UpdateProfile} from './components/UpdateProfile';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};