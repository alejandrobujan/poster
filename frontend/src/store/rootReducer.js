import {combineReducers} from 'redux';

import app from '../modules/app';
import users from '../modules/users';
import post from '../modules/post';
import catalog from '../modules/catalog';
import notification from '../modules/notification';

const rootReducer = combineReducers({
    app: app.reducer,
    users: users.reducer,
    post: post.reducer,
    catalog: catalog.reducer,
    notification: notification.reducer
});

export default rootReducer;