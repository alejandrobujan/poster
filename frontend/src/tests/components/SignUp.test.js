import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import {Provider} from 'react-redux';
import renderer from "react-test-renderer";
import store from "../../store";
import { SignUp } from "../../modules/users";

import {createMemoryHistory} from 'history';

describe("SignUp", () => {
	it("renders correctly", () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<MemoryRouter>
						<SignUp/>
					</MemoryRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

});