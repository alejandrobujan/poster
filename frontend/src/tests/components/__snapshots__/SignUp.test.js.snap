// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="card bg-light border-dark m-4"
  >
    <h5
      className="card-header text-center"
    >
      Sign up
    </h5>
    <div
      className="card-body"
    >
      <form
        className="needs-validation"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="login"
          >
            Login
          </label>
          <div
            className="col-md-6"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="login"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The login is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="email"
          >
            Email
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="email"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The email is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="password"
          >
            Password
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The password is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="confirmPassword"
          >
            Confirm password
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="confirmPassword"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The password confirmation is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="firstName"
          >
            First name
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="firstName"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The first name is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="lastName"
          >
            Last name
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="lastName"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The last name is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="avatar"
          >
            Avatar
          </label>
          <div
            className="col-md-6"
          >
            <input
              accept="image/*"
              id="avatar"
              onChange={[Function]}
              type="file"
            />
            <button
              className="btn btn-outline-danger"
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="button"
            >
              <svg
                className="bi bi-trash"
                fill="currentColor"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5Zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5Zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6Z"
                />
                <path
                  d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1ZM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118ZM2.5 3h11V2h-11v1Z"
                />
              </svg>
            </button>
            <div
              className="invalid-feedback"
            >
              The maximum size allowed is 1MB.
            </div>
          </div>
        </div>
        <div
          className="text-center"
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
